Dart/src/com/jetbrains/lang/dart/ide/errorTreeView/DartProblemsViewPanel.java

```diff
@@ -14,6 +19,13 @@ public DartProblemsViewPanel(Project project) {
   @Override
   protected void fillRightToolbarGroup(DefaultActionGroup group) {
     super.fillRightToolbarGroup(group);
+    group.add(new AnAction(DartBundle.message("dart.reanalyze.action.name"), DartBundle.message("dart.reanalyze.action.description"),
```

Comment:
I think action availability and behavior shouldn't differ whether it is invoked for Dart Analysis tool window, from main menu, from Find Action or from any custom location. If DAS works then action is available. Files open in editor or whatever else do not matter. Does it sound correct?
If so then just simplify ReanalyzeAction.update() and use PluginManager to add this action to tool window.